src/actions/preview.js

```diff
@@ -80,72 +108,44 @@ export function updatePreview(target: HTMLElement, editor: any) {
       }
     }
 
-    const source = getSelectedSource(getState());
-
-    const symbols = getSymbols(getState(), source.toJS());
-    if (symbols.identifiers.length == 0) {
+    if (isInvalidTarget(target)) {
       return;
     }
 
-    const invalidToken =
-      tokenText === "" || tokenText.match(/[(){}\|&%,.;=<>\+-/\*\s]/);
-
-    const invalidTarget =
-      (target.parentElement &&
-        !target.parentElement.closest(".CodeMirror-line")) ||
-      cursorPos.top == 0;
-
-    const isUpdating = preview && preview.updating;
+    if (!isLineInScope(getState(), tokenPos.line)) {
+      return;
+    }
 
-    const linesInScope = getInScopeLines(getState());
-    const inScope = linesInScope && linesInScope.includes(location.line);
+    const source = getSelectedSource(getState());
+    const symbols = getSymbols(getState(), source.toJS());
```

Comment:
hmmm im a bit surprised that getSymbols needs `toJS` -- is it because its used outside of the parser in this case?